src/RequestHandle.cpp

```diff
@@ -215,9 +179,10 @@ auto RequestHandle::SetVersion(
     }
 }
 
-auto RequestHandle::SetMaxBytes(size_t max_bytes) -> void
+auto RequestHandle::SetMaxDownloadBytes(ssize_t max_download_bytes) -> void
 {
-    m_max_bytes = max_bytes;
+    m_max_download_bytes = max_download_bytes;
+    m_bytes_written = 0;
```

Comment:
do you reset this field in Reset()?

Reply:
Ah, missed that!
Reset() now sets m_max_download_bytes to -1 (so it defaults to downloading everything) and m_bytes_written to 0.